version: '3.8'

services:
  postgres:
    image: postgres:14-alpine
    container_name: romaneio-postgres
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: admin
      POSTGRES_DB: romaneio_db
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
      - ./src/infrastructure/database/schema.sql:/docker-entrypoint-initdb.d/01-schema.sql
      - ./src/infrastructure/database/seed.sql:/docker-entrypoint-initdb.d/02-seed.sql
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5
    networks:
      - romaneio-network

  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: romaneio-api
    environment:
      DB_HOST: postgres
      DB_PORT: 5432
      DB_USER: postgres
      DB_PASS: admin
      DB_NAME: romaneio_db
      PORT: 3000
      NODE_ENV: development
      SWAGGER_ENABLED: true
    ports:
      - "3000:3000"
    depends_on:
      postgres:
        condition: service_healthy
    networks:
      - romaneio-network

volumes:
  postgres_data:

networks:
  romaneio-network:
    driver: bridge
